core/search/fields.py

```diff
@@ -145,16 +145,17 @@ def get_condition(self, expression: Expression, remainder: List[str]) -> Any:
 
         attribute_key = remainder[0]
 
-        # Checking permissions for a given key
-        if not g.auth_user.has_rights(Capabilities.reading_all_attributes):
-            metakey = (db.session.query(MetakeyPermission)
-                                 .filter(MetakeyPermission.key == attribute_key)
-                                 .filter(MetakeyPermission.can_read.is_(True))
-                                 .filter(g.auth_user.is_member(MetakeyPermission.group_id))).first()
-            if metakey is None:
-                raise ObjectNotFoundException(f"No such attribute: {attribute_key}")
-            if metakey.template.hidden and expression.has_wildcard():
-                raise FieldNotQueryableException("Wildcards are not allowed for hidden attributes")
+        metakey_definition = (
+            MetakeyDefinition.query_for_read(key=attribute_key, include_hidden=True).first()
+        )
+
+        if metakey_definition is None:
+            raise ObjectNotFoundException(f"No such attribute: {attribute_key}")
+
+        if (not g.auth_user.has_rights(Capabilities.reading_all_attributes)
+                and metakey_definition.hidden
+                and expression.has_wildcard()):
```

Comment:
change order maybe (move has_rights to the end)? If you explained it in english, you'd say
"Metakey is hidden, AND expression has a wildcard, BUT the user has no reading_all_attributes capability"